extensions/sensors/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+cmake_minimum_required(VERSION 2.6)
+
+
+set(CMAKE_EXE_LINKER_FLAGS "-Wl,--export-all-symbols")
+set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--export-symbols")
+
+include_directories(../../libminifi/include  ../../libminifi/include/core/yaml  ../../libminifi/include/core  ../../thirdparty/spdlog-20170710/include ../../thirdparty/concurrentqueue ../../thirdparty/yaml-cpp-yaml-cpp-0.5.3/include ${CIVET_THIRDPARTY_ROOT}/include ../../thirdparty/) 
```

Comment:
Not unique to this PR or ticket, but a lot of this extension boilerplate is really getting duplicated a lot... might be worth considering or adding a ticket for. (Not just this line but quite a bit of various duplicated cruft in an extension CMakeLists.txt).

Reply:
yeah I agree. I didn't take a healthy look at the PR before taking an experimental branch because I'm reasonably certain some of that cmake file isn't necessary. It would definitely help others to simplify some of that stuff. I'd also like to automate the acceptance of the extension dir and avoid people having to do more than define their dependencies for their extension -- i.e. not touching the root cmake.